ton_client/src/debot/routines.rs

```diff
@@ -1,24 +1,79 @@
 use super::dengine::TonClient;
-use crate::crypto::{signing_box_sign, ParamsOfSigningBoxSign, SigningBoxHandle};
-use crate::net::{query_collection, ParamsOfQueryCollection};
+use crate::crypto::{
+    generate_random_bytes, nacl_box_keypair_from_secret_key, signing_box_sign, KeyPair,
+    ParamsOfGenerateRandomBytes, ParamsOfNaclBox, ParamsOfNaclBoxKeyPairFromSecret,
+    ParamsOfSigningBoxSign, SigningBoxHandle,
+};
 use crate::encoding::{decode_abi_bigint, decode_abi_number};
+use crate::net::{query_collection, ParamsOfQueryCollection};
 use chrono::{Local, TimeZone};
 
 pub async fn call_routine(
     ton: TonClient,
     name: &str,
     arg: &str,
     signer: Option<SigningBoxHandle>,
-) -> Result<String, String> {
+) -> Result<serde_json::Value, String> {
+    let arg_json: Result<serde_json::Value, String> =
```

Comment:
Maybe better check the error at once? Not in every match

Reply:
There are two variants of argument for routine: as 1) 1 argument string and as 2) json string with args. `convertTokens` for example awaits float number encoded as string in `arg` parameter (`arg` is not a json in this case). And the same thing now with `loadBocFromFile`. In future it will be only 2nd variant. But now i support both to be compatible with existing debots.